code/contracts/ONEWallet2.sol

```diff
@@ -0,0 +1,176 @@
+
+                    if(!_executeTransfer(addr, amount)){
+                        lastRevealedOTPIdx = otpIdx; // do not allow to call reveal with this eotp again
+                        return false; // if daily limit is not met or there was an error with send()
+                    }                
+                    _cleanUpDailySpendings(); // clean up only after a successful transfer
+
+                } else if(operType == OperType.SET_LAST_RESORT_ADDR){
+                    if(lastResortAddress == address(0x0) && addr != address(0x0)) // alow setting up last resort address only if it was 0x0 before
+                        lastResortAddress = addr;
+                } else if(operType == OperType.RECOVERY){ // we do not care about params in Commits
+                    _drain(); // IH: I do not find any reason for this operation to be executed using OTP.
+                } else {
+                    emit UnknownOperationType(operType);                    
+                }                
+                
+                delete commits[otpIdx];
+                lastRevealedOTPIdx = otpIdx; // it is not possible to execute any operations using the otpIdx anymore (handled in commit)
+                return true;
+            }
+        }
+        return false;        
+    }
+
+    //////////////////////// Public Aux Functions ////////////////////////// 
+
+    /**
+     * It serves for the deletion of accidentaly submitted commits by the user or malicious commits submitted by the attackers, which contribute to the total storage allocation.
+     * Currently, this method can be called by anybody since it does touch only the garbage. 
+     * Also, in theory, someone may earn gas if there are too many entries submitted by the attacker (who payed for it).
+     */
+    function cleanUpCommits() external {                 
+        for (uint32 i = lastCleanUpOfCommitsIdx; i <= lastRevealedOTPIdx; i++){ // IH: in theory, this might lead to out-of-gas exception for settings of TOTP with very long validity but currently I'd not be affraid. Moreover, we are  getting the gas back for successfull deletions.
+            delete commits[i]; // NOTE: I'd say it makes no difference whether it was initialized or not, so we can delete any entry.        
+        }
+        lastCleanUpOfCommitsIdx = lastRevealedOTPIdx; // shift commit clean-up checkpoint
+    }
+
+    function getCurrentSpending() external view returns (uint){ 
+        return dailySpendings[block.timestamp / 1 days];
+    }        
+
+    function retire() external returns (bool){
+        require(uint32(block.timestamp / interval) - t0 > lifespan, "Too early to retire");        
+        return _drain();
+    }
+
+    //////////////////////// Internal Functions ////////////////////////// 
+
+    /**
+     * It cleans up the previous value of daily spending from the map (this ensures that map has always only a single value -- except the execution time of the reveal())
```

Comment:
This procedure essentially reduces dailySpendings to a single value, so it needs not to be a mapping anymore